rust/agama-manager/src/service.rs

```diff
@@ -679,7 +676,88 @@ impl MessageHandler<message::SolveStorageModel> for Service {
 impl MessageHandler<software::message::SetResolvables> for Service {
     /// It sets the software resolvables.
     async fn handle(&mut self, message: software::message::SetResolvables) -> Result<(), Error> {
+        self.check_stage(Stage::Configuring).await?;
         self.software.call(message).await?;
         Ok(())
     }
 }
+
+/// Implements the installation process.
+///
+/// This action runs on a separate Tokio task to prevent the manager from blocking.
+struct InstallAction {
+    l10n: Handler<l10n::Service>,
+    network: NetworkSystemClient,
+    software: Handler<software::Service>,
+    storage: Handler<storage::Service>,
+    files: Handler<files::Service>,
+    progress: Handler<progress::Service>,
+}
+
+impl InstallAction {
+    /// Runs the installation process on a separate Tokio task.
+    pub fn run(mut self) {
+        tokio::spawn(async move {
+            self.install().await.unwrap();
+        });
+    }
+
+    async fn install(&mut self) -> Result<(), Error> {
+        // NOTE: consider a NextState message?
+        self.progress
+            .call(progress::message::SetStage::new(Stage::Installing))
+            .await?;
+
+        //
+        // Preparation
+        //
+        self.progress
+            .call(progress::message::StartWithSteps::new(
+                Scope::Manager,
+                &[
+                    "Prepare the system",
```

Comment:
Are these user visible?
Translation infrastructure is still missing, but please mark these with gettextrs::gettext and we will make it work later.

Reply:
However, I am not sure about when to do the real translation. Let's think on a theorical case: you are installing the system, or doing the initial probing. You get the messages in, let's say, English. If you change the language, the rest of the process will be emitted in English.

I am tempted of translating the strings later (at API level) but it is true that, at least, we need to mark the strings here. WDYT?